pontoon/sync/core.py

```diff
@@ -20,6 +21,27 @@
 
 log = logging.getLogger(__name__)
 
+def run_only_one(func):
+    """
+    Decorator ensures that there's only one running task with given task_name.
+    """
+
+    task_name = "celery.{}".format(func.__name__)
+    def wrapped_func(*args, **kwargs):
+        # Moved import here to mock cache during tests.
+        from django.core.cache import cache
+
+        # Acquire the lock
+        if not cache.add(task_name, True):
```

Comment:
Yeah, but uncontrolled expire could spawn more processes that could stray lock... i could add send exception via RaygunApi to be notified about such things.